src/coreComponents/physicsSolvers/solidMechanics/SolidMechanicsEmbeddedFractures.cpp

```diff
@@ -845,8 +908,42 @@ void SolidMechanicsEmbeddedFractures::ApplyBoundaryConditions( real64 const time
                                           dofManager,
                                           localMatrix,
                                           localRhs );
+
+
+}
+
+void SolidMechanicsEmbeddedFractures::applyTractionBC( real64 const time_n,
+                                                       real64 const dt,
+                                                       DomainPartition & domain )
+{
+  FieldSpecificationManager & fsManager = FieldSpecificationManager::get();
+  // MeshLevel & mesh = *domain.getMeshBody( 0 )->getMeshLevel( 0 );
+
+  fsManager.Apply( time_n+ dt,
+                   &domain,
+                   "ElementRegions",
+                   viewKeyStruct::fractureTractionString,
+                   [&] ( FieldSpecificationBase const * const fs,
+                         string const & ,
+                         SortedArrayView< localIndex const > const & targetSet,
+                         Group * subRegion,
+                         string const &  )
+  {
+    GEOSX_UNUSED_VAR( targetSet );
+    SortedArray< localIndex > dummySet;
+    for (localIndex i=0; i < subRegion->size(); ++i)
+    {
+      dummySet.insert( i );
+    }
+    fs->ApplyFieldValue < FieldSpecificationEqual, parallelHostPolicy>( dummySet.toViewConst(),
+                                                                        time_n+dt,
+                                                                        subRegion,
+                                                                        viewKeyStruct::fractureTractionString);
+  });
+
 }
```

Comment:
Here, I want apply a user-defined value to the traction. This is useful, e.g., to run the Sneddon (pressurized fractured) problem without having to set up a poroelastic simulation. Basically, all I am doing is prescribing a b.c. for the traction (which I have now registered as a field) on the fracture surface. There is a small issue though that forced me to create this dummSet. The targetSet has size 0 because (I assume) it is created before the `EmbeddedSurfaceGenerator` has populated the `EmbeddedSurfaceSubRegion` so the subRegion is empty and consequently is the set "all". I have been able to understand how this set is created in the `FieldSpecificationManager`. 

@klevzoff  @rrsettgast , a little help? Once that's clear to me all I have to do it's modifying the set in the Generator as soon as the subRegion size is known. This is also useful to then prescribe an initial pressure or any other field in an existing fracture region since the same issue would occur.